### PR TITLE
Public reservation API

### PR DESCRIPTION
**Pull request by anzwdev**
A new public api class that can be used by other VS Code extensions to get IDs suggestions and to reserve them. There are 2 methods in the api
- `suggestIds` to get list of suggested IDs (1 for each ID range) without reserving them
- `reserverId` to reserve a single ID

``` javascript
let idNinjaExtension = vscode.extensions.getExtension("vjeko.vjeko-al-objid");
if ((idNinjaExtension) && (idNinjaExtension.isActive)) {
    let api = idNinjaExtension.exports;
    let ids = await api.suggestIds(uri, type);
}
```

### CODE REVIEW

**Comment by fvet** [comment]
@vjekob  Any chance in merging this PR into the AL Object Ninja master?

It would help us in adopting AZ AL Dev Tools combined with the power of AL Object ID Ninja ;)
https://github.com/anzwdev/al-code-outline/issues/276

**Comment by vjekob** [comment]
> @vjekob Any chance in merging this PR into the AL Object ID Ninja `master`?
> 
> It would help us in adopting AZ AL Dev Tools combined with the power of AL Object ID Ninja ;) [anzwdev/al-code-outline#276](https://github.com/anzwdev/al-code-outline/issues/276)

Yes, I am going to merge this PR and release a new version shortly.

File: vscode-extension/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "displayName": "AL Object ID Ninja",
     "description": "Manage object IDs in multi-user environments with mind-boggling simplicity.",
     "publisher": "vjeko",
-    "version": "2.9.2",
```
Comment (vjekob) [inline vscode-extension/package.json:6]: Please don't do this in future submissions. I'd like to control versioning of the extension myself. These kinds of changes are making my job more difficult.

**Review by vjekob — CHANGES_REQUESTED** [review]
Sorry, can you please exclude package-lock.json and package.json from this PR?

**Comment by vjekob** [comment]
I've decided to fix this myself as I want this feature to be a part of Ninja's next release.

Thanks for your contribution! 😊

**Comment by anzwdev** [comment]
That's great. Thank you :)

**Pull request merged**